roles/dbserver/tasks/main.yml

```diff
@@ -11,17 +11,26 @@
 - name: create db user
   sudo: yes
   sudo_user: postgres
-  postgresql_user: name={{ db_user }} password={{ db_pass }} role_attr_flags=SUPERUSER
+  postgresql_user: name={{ db_user }} password={{ db_pass }} role_attr_flags=CREATEDB,NOSUPERUSER
 
 - name: generate the .pgpass file
-  template: src=pgpass.j2 dest={{ home_path }}/.pgpass mode=600
+  template: src=pgpass.j2 dest={{ unicorn_home_path }}/.pgpass mode=600 owner={{ unicorn_user }}
+  sudo: yes
+  sudo_user: "{{ unicorn_user }}"
 
 - name: copy fix_pg_encoding PG script
-  copy: src=fix_pg_encoding.txt dest={{ home_path }}
+  copy: src=fix_pg_encoding.sql dest={{ unicorn_home_path }}
+  sudo: yes
+  sudo_user: "{{ unicorn_user }}"
   tags: pg-en
 
 - name: fix postgres db encoding problem
```

Comment:
Maybe there is a better way to call the script passed in one command instead of uploading, executing and deleting.